bioneuron_oracle/builder.py

```diff
@@ -29,6 +31,22 @@ def __init__(self):
         self.spikes = neuron.h.Vector()
         self.ap_counter.record(neuron.h.ref(self.spikes))
         self.num_spikes_last = 0
+        self._clean = False
+
+    def update(self):
+        if self._clean:
+            raise RuntimeError("cannot update() after cleanup()")
+        count = len(self.spikes) - self.num_spikes_last
+        self.num_spikes_last = len(self.spikes)
+        return count, np.asarray(self.v_record)[-1]
+
+    def cleanup(self):
+        if self._clean:
+            raise RuntimeError("cleanup() may only be called once")
+        self.v_record.play_remove()
```

Comment:
Ahhh, I've been searching for an appropriate NEURON method to remove vectors, and you finally found it! Do you remember the search terms/process you used to track down this function? I'm guessing you just went to the NEURON Vector documentation and searched it for 'delete', which seems obvious in retrospect...my google searches all led to random, mostly irrelevant posts on the NEURON forum.